### PR TITLE
Feature/red 325 student see grade

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: src/features/courses/course-controller.ts [hunk 1]
```diff
@@ -2236,12 +2247,80 @@ class CourseController {
                 END
             `);
 
-            attributes = [
-                [averageScoreAttribute, 'average'],
-                [sequelize.literal(pendingProblemCountCalculationString), 'pendingProblemCount'],
-                [sequelize.literal(masteredProblemCountCalculationString), 'masteredProblemCount'],
-                [sequelize.literal(inProgressProblemCountCalculationString), 'inProgressProblemCount'],
-            ];
+            // If the topicId isn't present (and implicitly, the questionId as well),
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:2250]: Why are we making this assumption? Why not show the open dead and closed score no matter what the filter is

Honestly, it is probably most important at the lowest filter level, just the course

File: src/features/courses/course-controller.ts [hunk 2]
```diff
@@ -2236,12 +2247,80 @@ class CourseController {
...
+            // include average grades for open-topics only and dead-topics only.
+            if (_.isNil(topicId)) {
+                // Calculate the OPEN grades only
+                const pointsEarnedOpen = `SUM(
+                    CASE
+                        WHEN "question".${CourseWWTopicQuestion.rawAttributes.optional.field} = FALSE
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:2256]: We haven't implemented yet but we should add a check for hidden false as well

File: src/features/courses/course-controller.ts [hunk 3]
```diff
@@ -2236,12 +2247,80 @@ class CourseController {
...
+                    END)`;
+                const averageScoreAttributeDead = sequelize.literal(`
+                    CASE WHEN ${pointsAvailableDead} = 0 THEN
+                        NULL
+                    ELSE
+                        ${pointsEarnedDead} / ${pointsAvailableDead}
+                    END
+                `);
+
+                attributes = [
```
Comment (PanopticaRising) [inline src/features/courses/course-controller.ts:2307]: Simplify this by pushing

File: src/features/courses/course-route.ts [hunk 4]
```diff
@@ -44,7 +45,12 @@ router.get('/statistics/units',
                 },
                 followQuestionRules: !_.isNil(req.query.userId)
             });
-            next(httpResponse.Ok('Fetched successfully', stats));
+
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            next(httpResponse.Ok('Fetched successfully', {
+                data: stats,
+                ...getAveragesFromStatistics(stats),
```
Comment (PanopticaRising) [inline src/features/courses/course-route.ts:52]: Create an averages property rather than spread.

**Review by doombot-exe — APPROVED** [review]

**Review by red-tt-bear — APPROVED** [review]

**Pull request merged**